### ISSUE TITLE
Failed Disk Shrink Error

### ISSUE THREAD

**Issue by mwech5191** [FSLogix/Invoke-FslShrinkDisk#60]
We have 152 VHDX profiles.  We started by testing individual profiles rather than running on the entire share.  We came across a few profiles that were erroring with the same error.  Over the weekend i ran the script on the entire share and it processed most of the profiles but we did see the same error on roughly 12 profiles.

DiskState column on csv stated "No Partition Supported Size Info - The Windows Disk SubSystem did not respond in a timely fashion try increasing number of cores or decreasing threads by using the ThrottleLimit parameter".

I have tried by changing the default value of the throttlelimit as stated in the error but have the same issue.  Is there anything we can do to resolve this or are we forced to create a new user profile?

Thanks,
Matt

**Comment by bw-ith** [comment]
@mwech5191  We were experiencing the same issue, we mounted the VHD's ran CHKDSK on them and only then after corruption was cleared were we able to run the FSLShrinkDisk script on these VHD's successfully. We're setting up a process to check the script completion log for these errors to ensure we fix the corruption before the profile hits the file max size by not shrinking.

**Comment by PyroSoul630** [comment]
For those of you getting issue with "Cannot get partition information"  due to IFS partition types: 

Please change this line of code in the script: 
from this: 

try {
                $allPartition = Get-Partition -DiskNumber $mountedDisk.Number -ErrorAction Stop

                if ($allPartition.Type -contains 'Basic') {
                    $partitionType = $true
                    $partition = $allPartition | Where-Object -Property 'Type' -EQ -Value 'Basic'
                }

try {
        $allPartition = Get-Partition -DiskNumber $mountedDisk.Number -ErrorAction Stop

        if ($allPartition.Type -contains 'Basic' -or $allPartition.Type -contains 'IFS') {
            $partitionType = $true
            $partition = $allPartition | Where-Object { $_.Type -eq "Basic" -or $_.Type -eq "IFS" }
        }


This literally took me years to figure out. Today was the day. Now all of my FSLogix VHDX files are finishing without error!! I'm so happy!

**Comment by b00walk** [comment]
@PyroSoul630 This should be committed to main branch, so simple and it works!

**Comment by PyroSoul630** [comment]
> @PyroSoul630 This should be committed to main branch, so simple and it works! Thank you!

I believe the original author has moved on to other things and also warned against modifying IFS file types. I haven't had a single issue with any of my user profiles thankfully. I would love the mention in the project but I'm just happy it works. Those file sizes get out of hand fast.

**Comment by lukaszdziwisz** [comment]
Hello @PyroSoul630, sorry to bring it up again but I have tried the new script and having an issue with shrinking individual VHDX files. Can you assist with that or point me in a correct direction if I'm doing something wrong?

Here is what I see when I try to shrink an individual profile:

.\Invoke-FslShrinkDisk.ps1 -Path D:\FSLogixProfiles\UserX\S-1-5-21-3438670015-1056806915-4004647813-12856_userx\Profile_userx.vhdx -Recurse -PassThru

EndTime        : 9:01:05 AM                                                                                     e -zing volume 1Running
ElapsedTime(s) : 153.8                                                                                                  umei             of
DiskState      : No Partition Supported Size Info - The Windows Disk SubSystem did not respond in a timely fashion try increasing number
                 cores or decreasing threads by using the ThrottleLimit parameter
OriginalSizeGB : 29.19
FinalSizeGB    : 29.19
FullName       : D:\FSLogixProfiles\userx\S-1-5-21-3438670015-1056806915-4004647813-12856_userx\Profile_userx.VHDX

When running the command I see a lot of 

Optimize-Volume -ErrorAction Stop 0% 0/1 completed
    Optimizing volume 100% Running 

and it keeps repeating like 100+ times

Any help would be greatly appreciated

**Comment by PyroSoul630** [comment]
Have you tried doing a dump of the stats on the VHDX? I am not the original creator. He has abandoned the project. I would check to see if the file has any white space in it. It may be erroring out because there's no space to free up. I forget the tool / command to check them but info should be out there. From the output, it seems like it's having a problem identifying the partition structure, which is what it did with IFS before I fixed the script. Does this happen on all your files or just this one? It may be corrupted. You might want to check your Citrix profile logs and see if there are any clues there. Next would be recreating the profile, I know everyone hates that, and see if that file is better, finishes compressing. 

Another option would be to dump the code into ChatGPT and the error output and see if it has any ideas. 

I'm thinking though that you probably just have a slightly corrupted file, not enough to break it but enough for the shrink function not to work. 

If I remember correctly the VHDX info tools will tell you whether the file is dirty, properly closed, or not. and quite a bit of other useful information. 

Good luck. reach out if you have any more questions and I'll try and help.

**Comment by lukaszdziwisz** [comment]
@PyroSoul630,

Thank you for quick response, I'll to research it more. It does happen on multiple profiles. Also, I'm seeing that FSlogix has new Compaction feature since version 2.9.x and higher so I'll try to test that and see if that helps

**Comment by PyroSoul630** [comment]
Pretty sure the compaction setting is on by default. I would definitely make sure you’re on the latest Fslogix. Check release notes. They are always fixing issues.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by lukaszdziwisz** [comment]
@PyroSoul630,

I agree that is what it says but looking at the registry I didn't see that key in place until I turn it on with GPO. We are currently on 3.25.401.153.05 but we did start well before 2.9.x so not too sure if that is why it's not there. Thank you

**Comment by PyroSoul630** [comment]
Yes, just configure in group policy and that should be it. You need to ensure the drfragsrvc is also enabled. Also, check to make sure the VHDX aren’t getting stuck open in the disk management. I’ve seen this happen in the past.

According to some internet posts it sounds like the script opens the disk and doesn’t assign the correct number or store it correctly and can’t find the disk number afterwards so it fails or loops. I don’t have a solution for this.

I would try using the command switches for the FSLSHRINK script. You can find them in the script itself or in the github. Set the throttle limit to 1 or better yet only point the script at a single file. Sometimes when it runs an entire directory, if you use that method, it gets hung up.


Thanks,

Michael Fahey,  IT Manager
Magna Health Systems
M. 312-676-0824  X. 4270

"If I cannot do great things, I can do small things in a great way." —Martin Luther King, Jr.

**Comment by JimMoyle** [comment]
This usually happend if you hve cretaed the disk using another method than FSLogix.  I decided when I wrote the script that I wouldn't be supporting any disks created by other tools than FSLogix as I couldn't predict the outcome.

You could also try repairing the disk before shrinking it.

FYI the reason I stopped updating this script is because disk compaction is now built into FSLogix and is supported directly by Microsoft.

I'll update the readme to reflect this as I'm of the opinion that people shouldn't really use the script anymore.

I decided to keep it up in case aaanyone needed to resuse any code.

**Comment by lukaszdziwisz** [comment]
@JimMoyle that sounds good, thank you for responding. I'll be testing the compaction provided by FSlogix and see how it goes.

**Comment by PyroSoul630** [comment]
@JimMoyle I appreciate you responding! I have never created an FSlogix container with any other tool and I have still had issues with compaction. That being said, I think some admins still find it useful to verify that FSlogix compaction is working correctly. It might seem outdated but I think there's still some value in having a secondary tool. If you are seeing errors with compaction and checking and seeing lots of whitespace this could be a solution to see if you can reclaim that space. Thanks for all your work on this tool Jim!
